src/commands/interface-manager/JSXFactory.ts

```diff
@@ -37,23 +37,51 @@ export function JSXFactory(tag: NodeTag, properties: unknown, ...rawChildren: (D
     return node;
 }
 
-
-// eslint-disable-next-line no-redeclare
-namespace JSXFactory {
-    export interface IntrinsicElements {
-        [elemName: string]: any;
-    }
-}
+// TODO: For `children?` to work without allowing people to accidentally use
+// `{undefined}` then we need to enable:
+// https://www.typescriptlang.org/tsconfig/#exactOptionalPropertyTypes
+// I do not know why this is not on by default when strict is true and
+// this is really annoying AaaAaaaAaaAaaAAaaAaa.
+// To enable exactOptionalPropertyTypes, we'll have to start with MPS
+// or extract DeadDocument into a library.
+type NodeProperties = { children?: JSXChild[]|JSXChild };
+type LeafNodeProperties = { children?: never[] };
 
 /**
- * Pisses me off that tooling is too dumb to use the above
+ * Has to be global for some reason or VSCodium explodes.
  * https://www.typescriptlang.org/docs/handbook/declaration-merging.html
  * https://www.typescriptlang.org/tsconfig#jsxFactory
  */
 declare global {
     export namespace JSX {
         export interface IntrinsicElements {
-            [elemName: string]: any;
+            a: NodeProperties & { href?: string, name?: string, target?: string },
+            b: NodeProperties,
+            br: LeafNodeProperties,
+            code: NodeProperties & { class?: string },
+            details: NodeProperties,
+            em: NodeProperties,
+            font: NodeProperties & { color?: string},
+            fragment: NodeProperties,
+            h1: NodeProperties,
+            i: NodeProperties,
+            li: NodeProperties,
+            ol: NodeProperties & { 'start'?: number },
+            p: NodeProperties,
+            pre: NodeProperties,
+            root: NodeProperties,
+            span: NodeProperties & {
+                'data-mx-bg-color'?: string,
+                'data-mx-color'?: string,
+                'data-mx-spoiler'?: string | null
```

Comment:
hmm, this needs fixing to be `data-max-spoiler?: string | undefined` now that i discovered https://www.typescriptlang.org/tsconfig/#exactOptionalPropertyTypes. But I wonder if there is another way?